From 0a4af7dcc865da8bf47476e3a8dc3b76f437af3e Mon Sep 17 00:00:00 2001
From: "Allen.Ye" <allen.ye@mediatek.com>
Date: Mon, 7 Aug 2023 15:27:27 +0800
Subject: [PATCH 38/69] mtk: hostapd: Fix rnr ie length when no need to report
 bss

Fix rnr ie length when no need to report bss. If we don't have content in
TBTT then don't change the length of the ie (*size_offset).

Signed-off-by: Allen.Ye <allen.ye@mediatek.com>
---
 src/ap/ieee802_11.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 8a26bdfbc..713b48433 100755
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -7732,8 +7732,10 @@ static u8 * hostapd_eid_rnr_iface(struct hostapd_data *hapd,
 		}
 
 		start = i;
-		*tbtt_count_pos = RNR_TBTT_INFO_COUNT(tbtt_count - 1);
-		*size_offset = (eid - size_offset) - 1;
+		if (tbtt_count != 0) {
+			*tbtt_count_pos = RNR_TBTT_INFO_COUNT(tbtt_count - 1);
+			*size_offset = (eid - size_offset) - 1;
+		}
 	}
 
 	if (tbtt_count == 0)
-- 
2.39.2

